### PR TITLE
Keysight J7211 attenuator

### PR DESCRIPTION
**Pull request by mgunyho**
Very simple driver for the Keysight J7211 attenuator. There's no example notebook because the driver is so basic, is that okay?

### CODE REVIEW

File: qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    Args:
+        name: Instrument name
+        address: Address or VISA alias of instrument
+        attenuation: Optional attenuation level (in dB) to set on startup
+    """
+
+    def __init__(self, name: str, address: str,
+                 attenuation: Optional[int] = None, **kwargs):
+        super().__init__(name=name, address=address, terminator='\r', **kwargs)
```
Comment (astafan8) [inline qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py:19]: please put the terminator to the init of the driver such that it's possible to change it if needed.
```suggestion
    def __init__(self, name: str, address: str,
                 attenuation: Optional[int] = None, terminator='\r', **kwargs):
        super().__init__(name=name, address=address, terminator=terminator, **kwargs)
```

File: qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        if model in ["J7211A", "J7211B"]:
+            self.attenuation.vals = Ints(0, 120)
+        elif model in ["J7211C"]:
+            self.attenuation.vals = Ints(0, 100)
+        else:
+            raise RuntimeError(f"Model {model} is not supported.")
+
+        if attenuation is not None:
+            self.attenuation(attenuation)
```
Comment (astafan8) [inline qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py:38]: this feature is supported by Parameter's `initial_value` argument

File: qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    def __init__(self, name: str, address: str,
+                 attenuation: Optional[int] = None, **kwargs):
+        super().__init__(name=name, address=address, terminator='\r', **kwargs)
+
+        self.add_parameter('attenuation', unit='dB',
+                           set_cmd='ATT {:03.0f}',
+                           get_cmd='ATT?',
+                           get_parser=int)
+
+        self.connect_message()
```
Comment (astafan8) [inline qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py:26]: it's recommended put this at the end of the init for consisency with other drivers

File: qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        model = self.IDN()['model']
+
+        if model in ["J7211A", "J7211B"]:
+            self.attenuation.vals = Ints(0, 120)
+        elif model in ["J7211C"]:
+            self.attenuation.vals = Ints(0, 100)
```
Comment (astafan8) [inline qcodes_contrib_drivers/drivers/Keysight/Keysight_J7211.py:33]: it might be more logical to create validators BEFORE creating the parameter.

**Comment by mgunyho** [comment]
Thanks for the feedback! The changes are implemented in a268e6f.

**Pull request merged**